flexboxAPIspecV2.md

```diff
@@ -0,0 +1,783 @@
+    each with an API page, and then there's also a conceptual page that
+    discusses them collectively
+    (https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/text-controls).
+
+    Another way to use this section is as a draft of a blog post that introduces the new feature.
+
+    Sometimes it's difficult to decide if text belongs on a how-to page or an API page.
+    It's not important to make a final decision on that in this spec; we can always
+    adjust it when copying to DMC.
+-->
+
+WinUI Offers several Layout options, such as StackLayout and Flowlayout. Usage of Flexbox Layout will behave similarly and interact by wrapping your desired elements in a Layout Control.
+
+#### Flex Containers vs Flex Items
+
+Flex Containers are elements that hold and specify the arangement of Flex Items. Containers have the following accepted properties:
+| Property | 
+|-|
+| Direction |
+Wrap|
+JustifyContent|
+AlignItems|
+AlignContent|
+
+Flex Items are elements that are children of a Flex Container. Items have the following accepted properties:
+| Property | 
+|-|
+order|
+Grow|
+Shrink|
+Basis|
+AlignSelf|
+Unit|
+
+
+
+#### Traditional CSS Behavior vs Flexbox in WinUI
+
+Several behaviors may deviate from the exact behavior found in CSS Flexbox, to highlight one in XAML, items do not overflow their containers, instead they will be omitted from the layout.
+
+
+
+# API Pages
+
+_(Each of the following L2 sections correspond to a page that will be on docs.microsoft.com)_
+
+<!-- TEMPLATE
+
+  Each of the L2 sections in this "API Pages" section corresponds to a page on DMC.
+
+  It's not necessary to have a section for every class member though:
+  * If its purpose and usage is obvious from it's name/type, it's not necessary to
+    create a section for it.
+  * If its purpose and usage is fully explained by brief description, either
+      put it in a table in the "Other [class] members" section
+      put it with /// comments in the IDL section
+
+  Create an L2 section here for each API that needs more description or examples.
+  For a new class with members, the members should go in their own L2 section.
+
+  Example layout
+    ## MyClass
+    ## MyClass.Member1
+    ## MyClass.Member2
+    ## Other MyClass members
+    ## MyOtherClass
+    ## ...
+
+  Notes:
+  * The first line of each of these sections should become that first line on the DMC page,
+    which then becomes the description you see in Intellisense.
+  * Each page can have description, examples, and remarks.
+    Remarks are where the documentation calls out special considerations that the developer
+    should be aware of.
+  * It can be helpful at the top of an API page (or after the Intellisense text) to add the
+    API signature in C#
+  * Add a "_Spec note: ..._" to add a note that's useful in this spec but shouldn't go to DMC.
+  * Show _examples_, not _samples_; an example is a snippet, a sample is a full working app.
+
+-->
+
+## Flex Direction
```

Comment:
```suggestion
## FlexDirection enum
```

Each of these L2 sections in the API Pages should be an API name. We don't need to have a section for every API though because some are obvious. For example once we describe FlexDirection here we don't need to also have a section for the FlexDirection property; just having that property in the IDL is good enough.